tests/llmcompressor/test_kv_cache.py

```diff
@@ -0,0 +1,310 @@
+import os
+import tempfile
+
+import pytest
+from compressed_tensors.compressors import ModelCompressor
+from compressed_tensors.quantization.cache import KVCacheScaleType
+from compressed_tensors.quantization.utils.helpers import iter_named_quantizable_modules
+from datasets import load_dataset
+from transformers import AutoTokenizer
+
+from llmcompressor.transformers import SparseAutoModelForCausalLM, oneshot
+
+MODEL_IDS = [
+    "TinyLlama/TinyLlama-1.1B-Chat-v1.0",
+    "microsoft/Phi-3-mini-4k-instruct",
+]
+
+
+@pytest.fixture(scope="session")
+def oneshot_fixture():
+    num_bits = 8
+    _type = "float"
+    strategy = "tensor"
+    dynamic = False
+    symmetric = True
+
+    recipe = f"""
+    quant_stage:
+        quant_modifiers:
+            QuantizationModifier:
+                kv_cache_scheme:
+                    num_bits: {num_bits}
+                    type: {_type}
+                    strategy: {strategy}
+                    dynamic: {dynamic}
+                    symmetric: {symmetric}
+    """
+
+    used_args = dict(
+        num_bits=num_bits,
+        _type=_type,
+        strategy=strategy,
+        dynamic=dynamic,
+        symmetric=symmetric,
+    )
+
+    oneshot_args = dict(
+        dataset="open_platypus",
+        recipe=recipe,
+        num_calibration_samples=16,
+    )
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        for model_id in MODEL_IDS:
+            oneshot_args["output_dir"] = os.path.join(tmpdir, model_id)
+            used_args["output_dir"] = oneshot_args["output_dir"]
+            yield oneshot(model=model_id, **oneshot_args), used_args
```

Comment:
Why not cache the results of this generator? Then you can test the results of each model with each test, rather than only testing the first model with the first test and the second model with the second test

Reply:
oh its does that already

Reply:
@horheynm As I read it, the tests which use this feature only run with one model each (for example, `test_kv_cache_config_format` only runs with the `TinyLlama/TinyLlama-1.1B-Chat-v1.0` model. Is that incorrect?

Reply:
I just confirmed through debugging that `test_kv_cache_config_format` only runs with `TinyLlama/TinyLlama-1.1B-Chat-v1.0`. What I'm suggesting is maybe you could cache the oneshot results and then you'd be able to run each of your tests with each of the models